client/packages/common/src/utils/BarcodeScannerContext.tsx

```diff
@@ -84,7 +91,18 @@ export const BarcodeScannerProvider: FC<PropsWithChildrenOnly> = ({
     if (hasElectronApi) {
       try {
         const { startBarcodeScan } = electronNativeAPI;
-        const data = await startBarcodeScan();
+        startBarcodeScan();
+        const scan = new Promise<number[]>((resolve, reject) => {
+          electronNativeAPI.onBarcodeScan((_event, data) => {
+            try {
+              resolve(data);
```

Comment:
Is this try catch statement correct ? (promise should ether resolve or reject not both), if try catch was wrapping `electronNativeAPI.onBarcodeScan` it would make a little bit more sense, although I think this should just resolve(data) without reject ?

Reply:
It seemed like a neat way to handle the error, but I agree. Shouldn't be trying to resolve and reject. I'll remove in #1629